projectlist/api.py

```diff
@@ -20,3 +22,27 @@ def get_queryset(self):
         return PersonalProject.objects.none()
       else:
         return PersonalProject.objects.filter(user=user)
+
+class UserSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = User
+        fields = '__all__'
+    def create(self, validated_data):
+        user = User.objects.create_user(validated_data['username'])
+        user.set_password(validated_data['password'])
+        user.user_permissions.add(Permission.objects.get(name='Can add personal project'))
+        user.user_permissions.add(Permission.objects.get(name='Can change personal project'))
+        user.user_permissions.add(Permission.objects.get(name='Can delete personal project'))        
+        user.user_permissions.add(Permission.objects.get(name='Can view personal project'))
+        user.save()
+        return user
+
+
+class CreateUserView(viewsets.ModelViewSet):
+    model = User
+    queryset = User.objects.none()
+
+    permission_classes = [
+        permissions.AllowAny # Or anon users can't register
+    ]
+    serializer_class = UserSerializer
```

Comment:
Nice job on this - making sure that new users can sign up is important! And your permissions for each of the CRUD functions look great.